### PR TITLE
Review and implement seaside restoration model #47

### PR DESCRIPTION
**Pull request by longshuicy**
This is ready to be reviewed. Just run the `tests/pyincore/analyses/indp/test_indp.py`. It has chained analyses; but some from production some from dev. Will clean them up with a jupyter notebook in incore-doc soon

### CODE REVIEW

File: pyincore/analyses/indp/dislocationutils.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                        values = [t, n, val['total_pop_node'] - val['num_dilocated'][t],
+                                  val['total_pop_node']]
+                        dynamic_params[nn] = dynamic_params[nn].append(dict(zip(dp_dict_col, values)),
+                                                                       ignore_index=True)
+        with open(output_file, 'wb') as f:
+            pickle.dump(dynamic_params, f)
+        return dynamic_params
+
+    @staticmethod
+    def lumberton_disloc_time_mode(household_data):
```
Comment (navarroc) [inline pyincore/analyses/indp/dislocationutils.py:132]: is this specific to lumberton?

Comment (longshuicy) [reply pyincore/analyses/indp/dislocationutils.py:132]: oh noes i didn't notice this module... 
this looks like something we should separate out. Maybe in the next release?

Comment (longshuicy) [reply pyincore/analyses/indp/dislocationutils.py:132]: for now i can rename this just `disloc_time_mode`?

Comment (navarroc) [reply pyincore/analyses/indp/dislocationutils.py:132]: Sounds good. Refactoring next release works so we can get this in.

File: pyincore/analyses/indp/indpcomponents.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2019 University of Illinois and others. All rights reserved.
```
Comment (navarroc) [inline pyincore/analyses/indp/indpcomponents.py:1]: 2023

**Comment by ylyangtw** [comment]
The first time I tested it, I got `No executable found for solver 'scip'`. 
So I installed scip using pip, but got `subprocess.TimeoutExpired: Command '['/opt/homebrew/anaconda3/envs/pyincore310/bin/scip', '--version']' timed out after 1 seconds`

**Comment by longshuicy** [comment]
> The first time I tested it, I got `No executable found for solver 'scip'`. So I installed scip using pip, but got `subprocess.TimeoutExpired: Command '['/opt/homebrew/anaconda3/envs/pyincore310/bin/scip', '--version']' timed out after 1 seconds`
> 
> I also tried reinstalling requirements.txt but didn't get any luck yet...
can you check is scip is in your conda environment? 
If not you need to install `scip` by `mamba install scip`

If scip is there, try `which scip` and see which path it's point towards?

**Review by navarroc — APPROVED** [review]

**Review by ylyangtw — APPROVED** [review]
Changes look good. Test has been running over 20 mins without problems.
Maybe add a change log?

**Pull request merged**